packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -830,6 +840,64 @@ class _SelectableRegionState extends State<SelectableRegion> with TextSelectionD
     await Clipboard.setData(ClipboardData(text: data.plainText));
   }
 
+  bool? _adjustingSelectionExtent;
+  bool _determineIsAdjustingSelectionExtend(bool forward) {
+    if (_adjustingSelectionExtent != null) {
```

Comment:
The cache is here so that we will be moving the same selection handle once decided until user starts a new selection(a clear selection will be called).

Reply:
I am not sure if I understand. why end always represent the active edge given that start can be larger or smaller?

Reply:
According to https://api.flutter.dev/flutter/services/TextSelection/extent.html 
> When the user uses the arrow keys to adjust the selection, this is the value that changes. 

Does it make sense to do the same here?

Reply:
Is the document up to date? I thought even for TextField this may not be the case, it will pick the one that will expand the selection?